src/@3rdweb-sdk/react/hooks/useEngine.ts

```diff
@@ -33,6 +33,12 @@ export type EngineInstance = {
     | "deploymentFailed";
 };
 
+const getEngineRequestHeaders = (token: string | null): HeadersInit => ({
+  "Content-Type": "application/json",
+  "ngrok-skip-browser-warning": "true",
+  Authorization: `Bearer ${token}`,
```

Comment:
If token is null we should omit this header.
```suggestion
  Authorization: token ? `Bearer ${token}` : undefined,
```